cpp/src/command_classes/DoorLock.cpp

```diff
@@ -194,6 +206,11 @@ namespace OpenZWave
 						value->OnValueRefreshed(lockState);
 						value->Release();
 					}
+					if (Internal::VC::ValueList* value = static_cast<Internal::VC::ValueList*>(GetValue(_instance, ValueID_Index_DoorLock::Door_Condition)))
+					{
+						value->OnValueRefreshed(_data[3]);
+						value->Release();
+					}
```

Comment:
Few issues here:
1) this ValueID is not created by the Class. You need to determine if the device in question supports this ValueID (probably based upon the CommandClass Version) and then create the ValueID, and subsequently, in this specific block, only do this update if the CommandClass Version is equal/above the minimum version and you should also check if index 3 is valid in the data recieved.

Reply:
I was looking through the ZW docs and it appears DoorCondition is defined as far back as version 1 of the protocol and hasn't changed sense. I'm assuming this negates the need for any version checking? I added a simple validation of index 3.